packages/edit-site/src/components/navigation-sidebar/navigation-toggle/index.js

```diff
@@ -35,6 +35,16 @@ function NavigationToggle( { icon, isOpen } ) {
 
 	const { setIsNavigationPanelOpened } = useDispatch( editSiteStore );
 
+	document.addEventListener(
```

Comment:
Instead of adding an event listener on the document, we should probably contain this to the wrapper of the panel itself.  A recent example could be with the new list-view component:

https://github.com/WordPress/gutenberg/blob/5ae2b1b06666ad1a75d5e92bfbdec4b4e728b071/packages/edit-site/src/components/secondary-sidebar/list-view-sidebar.js#L60

https://github.com/WordPress/gutenberg/blob/5ae2b1b06666ad1a75d5e92bfbdec4b4e728b071/packages/edit-site/src/components/secondary-sidebar/list-view-sidebar.js#L45

Something like that should work well here also.